ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -293,16 +288,12 @@ static map_parse_frame_t* map_push_frame(VALUE map,
   native_slot_init(handlerdata->key_field_type, &frame->key_storage);
   native_slot_init(handlerdata->value_field_type, &frame->value_storage);
 
-  rb_ary_push(map_parse_frames,
+  Map_push_frame(map,
```

Comment:
If you are storing this in the map directly, I don't think it needs to be an array with push and pop. I think it could just be a single value. The array was for when we had a single global stack that needed to be able to handle parsing of nested maps.

Reply:
Ok, I changed this to store only one value.